2022/Q2/artifacts/pos-model/PoS-model-cubic.md

```diff
@@ -0,0 +1,493 @@
+## Assumptions/simplifications
+
+- There is a single token type
+
+## Data types
+
+```go
+type Addr
+type Key
+type Epoch uint
+type VotingPower uint
+
+type Validator struct {
+  consensus_key map<Epoch, Key>
+  state map<Epoch, {inactive, candidate, frozen, jailed}>
+  total_deltas map<Epoch, amount:int>
+  voting_power map<Epoch, VotingPower>
+  reward_address Addr
+}
+
+type Bond struct {
+  validator Addr //not used
+  source Addr //not used
+  deltas map<Epoch, int>
+  }
+
+type Unbond struct {
+  validator Addr //not used
+  source Addr //not used
+  deltas map<(start:Epoch, end:Epoch), int>
+}
+
+type Slash struct {
+  epoch Epoch
```

Comment:
Do we need this? (it is now the key of the map containing this structure)

Reply:
Actually, there are some inconsistencies. Withdrawing is till using the slashes variables indexed by validators. I'll fix this and remove the epoch field if it is not needed. Thanks for pointing it out!

Reply:
I think I fixed it.